autosubmit/job/job_packages.py

```diff
@@ -136,27 +137,56 @@ def _create_scripts_threaded(self, jobs: list[Job], configuration: 'AutosubmitCo
+            return
+
+        Log.debug("Checking job files existence")
+        project_dir = Path(configuration.get_project_dir())
+        # This self.jobs is actually only jobs in the package not all the loaded job_list.
+        for job in self.jobs:
+            # When SCRIPT is defined, FILE is ignored.
+            if not job.script and job.file:
+                job_file_path = project_dir / job.file
+                if not job_file_path.exists():
+                    if only_generate:
+                        Log.warning(f"[section:{job.section}]: Job script:{job.file} does not exists, skipping check")
+                    else:
+                        raise AutosubmitCritical(f"[section:{job.section}]: Job script:{job.file} does not exists", 7014)
+
+            for additional_file in job.additional_files:
+                exists = not additional_file or (project_dir / additional_file).exists()
+                if not exists and only_generate:
+                    Log.warning(f"[section:{job.section}]: Additional file:{additional_file} does not exists, skipping check")
+                elif not exists and not only_generate:
+                    raise AutosubmitCritical(f"[section:{job.section}]: Additional file:{additional_file} does not exists", 7014)
+
+    def build_scripts(self, configuration: 'AutosubmitConfig') -> None:
+        """Submit jobs one by one without using threads.
+
+        :param configuration: Autosubmit basic configuration.
+        :type configuration: AutosubmitConfig
+        :return: None.
         """
+        Log.debug("Checking Scripts")
         for job in self.jobs:
-            if only_generate and not os.path.exists(os.path.join(configuration.get_project_dir(), job.file)):
-                break
-            if not os.path.exists(os.path.join(configuration.get_project_dir(), job.file)):
-                if configuration.get_project_type().lower() != "none" and len(configuration.get_project_type()) > 0:
-                    raise AutosubmitCritical(
-                        "Template [ {0} ] using CHECK=On_submission has some empty variable {0}".format(job.name), 7014)
-            if not job.check_script(configuration, show_logs=job.check_warnings):
-                Log.warning(
-                    f'Script {job.name} has some empty variables. An empty value has substituted these variables')
-            else:
-                Log.result("Script {0} OK", job.name)
-            # looking for directives on jobs
-            self._custom_directives = self._custom_directives | set(job.custom_directives)
+            if job.file or job.script:
+                if not job.check_script(configuration, show_logs=job.check_warnings):
+                    Log.warning(
+                            f"Script {job.name} has some empty variables. An empty value has substituted these variables"
+                        )
```

Comment:
I believe we could simplify the code by doing this
```suggestion
            if (job.file or job.script) and not job.check_script(configuration, show_logs=job.check_warnings):
                Log.warning(
                    f"Script {job.name} has some empty variables. An empty value has substituted these variables")
```